files/instanceSetupLambda.js

```diff
@@ -21,6 +21,12 @@ function recordCompletion(event, success) {
 
 exports.handler = async (event) => {
   const instanceId = event['EC2InstanceId'];
+  /*
+   * Possible deployment types include:
+   * - 'setup': Full end-to-end configuration, from vanilla operating system to live instance.
+   * - 'deploy': Take a vanilla operating system and set up a warm instance, but do not make it active.
+   * - 'configure': Take a warm instance and make it active.
+   */
```

Comment:
when i see the options of 
setup, deploy, configure.

I do not envision 'setup' doing everything.  nor 'deploy' only making a 'warm' instance.

@chris-randall-qol and others,  what is your thoughts on this. 

I know this is from legacy deployPhase's we got from inheriting this codebase but i think its time to make it less confusing.

Reply:
> I know this is from legacy deployPhase's we got from inheriting this codebase

Yes, those were the OpsWorks terms. We _could_ change them, it would just mean a lot of renaming.

Note that the distinction between 'deploy' and 'configure' creating warm vs hot instances did not come from OpsWorks. It's just something that I've observed to be useful if we want to make custom images.